neat_cache/lib/src/providers/resp.dart

```diff
@@ -0,0 +1,514 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Minimalistic [RESP[1] protocol implementation in Dart.
+///
+/// [1]: https://redis.io/topics/protocol
+library resp;
+
+import 'dart:async';
+import 'dart:collection';
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:logging/logging.dart';
+
+final _log = Logger('neat_cache:redis');
+
+/// Thrown when a command returns
+class RedisErrorException implements Exception {
```

Comment:
This name is a bit strange, why not just `RedisException` or maybe `RedisServerException`?

Reply:
hmm, maybe `RedisCommandException` is better, because it due to an error from a command usually...

Reply:
sgtm